app/models/decidim/kids/minor_account.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Kids
+    # Abstract class from which all models in this engine inherit.
+    class MinorAccount < ApplicationRecord
+      self.table_name = "decidim_kids_minor_accounts"
+
+      belongs_to :tutor,
+                 foreign_key: "decidim_tutor_id",
+                 class_name: "Decidim::User"
+      belongs_to :minor,
+                 foreign_key: "decidim_minor_id",
+                 class_name: "Decidim::User"
+
+      validate :same_organization
+      validate :can_be_tutor
+      validate :can_be_minor
+
+      private
+
+      def same_organization
+        return if tutor.try(:organization) == minor.try(:organization)
+
+        errors.add(:tutor, :invalid)
+        errors.add(:minor, :invalid)
+      end
+
+      def can_be_tutor
+        return unless tutor.minor? || !tutor.confirmed?
+
+        errors.add(:tutor, :invalid)
+      end
+
+      def can_be_minor
+        return unless minor.tutor? || minor.admin?
```

Comment:
Maybe it's not time to take into account it. But, what if the minor becomes an adult and wants to create minor accounts? the `minor.tutor?` will return true because there are existing tutor accounts.

Reply:
I think we shouldn't care about this here.
The process to convert a minor to and adult will have to remove their entry as a minor_account, so it won't be tutelated anymore.
This method is here to ensure nor admins or existing tutors can be converted to a minor account, that would be a security concern in my opinion. Anyway, we don't expect existing users to be converted to minors, only new accounts so this is just a safeguard.

Reply:
Ok!